main.cpp

```diff
@@ -1,25 +1,97 @@
+
+        opt = getopt_long(argc, argv, "he:d:f:", long_options, &option_index);
+
+        switch (opt) {
+            case 'h':
+                printf("Usage: ex1 [OPTION]...[\"DIRECTORY\"]..\n"
+                       "Used to encrypt and decrypt directories given a root path as the argument. You can only\n"
+                       "use on argument at a time encrypt or decrypt\n\n"
+                       "options:\n"
+                       "-e, --encrypt               | Encrypts files in a given directory and accepts the root path as an argument \n"
+                       "-d, --decrypt               | Decrypts encrypted files from this program and accepts the root path as an argument\n"
+                       "-f                          | Checks for duplicate files in a directory\n"
+                       "-h, --help                  | Prints help message\n"
+                );    //placeholder
+                break;
+            case 'e':
+                directory = optarg;
+                if (eord < 0) {
+                    eord = 0;
+                }else{
+                    eord = 2;
+                }
+                break;
+            case 'd':
+                directory = optarg;
+                if (eord < 0) {
+                    eord = 1;
+                }else{
+                    eord = 2;
+                }
+                break;
+            case 'f':
+                directory = optarg;
+                if (eord < 0) {
+                    eord = 3;
+                }else{
+                    eord = 2;
+                }
+                break;
+            case 1:
+                printf("no arguments\n");    //default non-argument case
+                break;
+        }
+    }
+
     getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
 
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
+    if(eord == 0){
+        for (fs::path &file: files) {
+            encrypt(file);
+            std::cout << "Encrypted " << file.string() << "\n";
+        }
+    }else if(eord == 1){
+        for (fs::path &file: files) {
+            decrypt(file);
+            std::cout << "Decrypted " << file.string() << "\n";
+        }
+    }else if(eord == 2){
+        std::cout << "Must choose either encrypt or decrypt or file duplication check" << std::endl;
+    }else if(eord ==3){
+
+        for(fs::path &file1: files){
+            for(fs::path &file2: files){
+                if(file1 == file2) //compares file objects to ensure it doesn't compare to itself.
+                    break;
+                if(is_data_duplicate(file1, file2)){
+                    std::cout << file1.string() << " and " << file2.string() << " are duplicates.\n";
+                }
+            }
+        }
```

Comment:
This loop should be optimized a la `j = i + 1`
```cpp
for(int i=0;i<N;++i)
    for(int j=i+1;j<N;++j)
        process_pair( X[i], X[j] );
```

Reply:
You're welcome to try a more advanced C++ implementation of it.. but I honestly indexing the vector is totally fine.
https://cwzx.wordpress.com/2014/04/15/iterating-over-distinct-pairs-in-cpp/

Reply:
Yeah I didn’t have much time to look into it  so just went with the easiest implementation I knew of but I’ll look at that for the future could be super useful!

Reply:
I’ll optimize it tonight when I get back but it does work as is for now

Reply:
for loop optimized